config/routes.rb

```diff
@@ -77,7 +77,11 @@
         json_api_resources :pages, controller: "project_pages"
       end
 
-      json_api_resources :workflows, links: [:subject_sets], versioned: true
+      json_api_resources :workflows, links: [:subject_sets], versioned: true do
+        member do
+          post :retire_subject
```

Comment:
Not sure i'm following 100%, I think your reasoning is questioning the situation where we have a `workflow_id` and a `subject_id` but we don't know the implementation details of the `set_member_subject_id` so we can't just append to the SMS `retired_workflow_ids` array.

This is due to the fact that the client applications deal in subject ids (raw media locations, metadata, etc) and the internal SubjectSet, SetMemberSubject models are really just how we implmenet the links between projects & subjects. 

So your retire end points would need to determine the internal link models to retire and update the SMS `retired_workflow_ids` array. Make sense? PM me if you want to chat more about this.